docs/src/further_topics/ugrid/partner_packages.rst

```diff
@@ -0,0 +1,94 @@
+need unworkably large time/resource to perform on an unstructured mesh. New
+approaches are needed...
+
+Python packages already exist that enable the use of powerful mesh analysis
+tools originally written in faster languages such as C. And the Iris team are
+working on wrapper solutions to 'bridge the gap' between these packages and
+Iris.
+
+These solutions have been placed in their own 'partner packages' for several
+reasons:
+
+* Can be useful to others who are not using Iris.
+
+  * Everyone working with multi-dimensional geographic datasets shares common
+    problems that need solving.
+  * Wider user base = stronger community = better solutions.
+
+* Only some Iris users will need them - they are **optional** Iris dependencies.
+
+  * They introduce a lot of new API.
+  * They introduce new large dependencies that take time to install and need
+    disk space.
+
+Below you can learn more about the partner packages and how they are useful.
+Specifics of what operations would require their installation can be found in:
+:doc:`operations`.
+
+.. important:: **Experimental**
+
+        As with Iris' mesh support, these packages are still in the
+        experimental stages. They would love your feedback, but as immature
+        packages their API, documentation, test coverage and CI are still
+        'under construction'.
+
+
+`GeoVista`_
+===========
+.. image:: images/geovistalogo.svg
+   :width: 300
+   :class: no-scaled-link
+
+.. rubric:: "Cartographic rendering and mesh analytics powered by `PyVista`_"
+
+PyVista is described as "VTK for humans" - VTK is a very powerful toolkit for
+working with meshes, and PyVista brings that power into the Python ecosystem.
+GeoVista in turn makes it easy to use PyVista specifically for cartographic
+work, designed from the start with the Iris
+:class:`~iris.experimental.ugrid.Mesh` in mind.
+
+Applications
+------------
+* Interactively plot mesh data:
+
+  * On a 3D globe.
+  * On your favourite projection.
+
+* Extract a specific region from a mesh.
+* Combine multiple meshes into one.
+
+`iris-esmf-regrid`_
+===================
+.. image:: images/iris-esmf-regrid.svg
+   :width: 300
+   :class: no-scaled-link
+
+.. rubric:: "A collection of structured and unstructured ESMF regridding schemes for Iris"
+
+ESMF provide a sophisticated, performant regridding utility that supports a
+variety of regridding types with both structured grids and unstructured meshes,
+and this also has a flexible Python interface - ESMPy. iris-esmf-regrid takes
+advantage of having a specific use-case - regridding Iris
+:class:`~iris.cube.Cube`\s - to provide ESMPy-Iris wrappers that make the
+process as easy as possible, with highly optimised performance.
+
+Applications
+------------
+* Regrid structured to unstructured.
+* Regrid unstructured to structured.
+* Regrid while maintaining data laziness.
+* Save a prepared regridder for re-use in subsequent runs.
```

Comment:
For me, I think the important feature of iris-esmf-regrid over plain ESMF is not just that the regridders are re-usable, but they are also faster.  Because IIUC, we use ESMF to calculate weights, but we can perform the lazy array multiplications faster elsewhere.
Does this make sense @stephenworsley ?

Reply:
I have made some mentions of the performance enhancements, but perhaps not enough. I'm totally happy to change if anyone can suggest the specific phrasing that would be better.

Reply:
So I don't think I could say in full confidence that there is a performance improvement in every case. ESMF seems to have its own way to handle parallelism which I could never quite figure out. I will say that there are difinitely circumstances where we are able to provide significantly better performance. I think one of the biggest improvements is the fact that a regridder can be reused on sources with different masks, where in ESMF you would have to recalculate the weights every time. I believe that regridding sources with large non-horizontal coordinates also performs better (at the moment, this is the only case where parallelism is possible), though performance here also depends on how the array is chunked.

In terms of what is worth adding here as a feature, I think "Regrid while maintaining data laziness" already sort of implies dask integration. Maybe the performance could be emphasised as: "Regrid with dask integration, computing in parallel and maintaining data laziness."
The handling of masked data could also be emphasised with a bullet point something like: "Improved handling of masked data", If you think this is the place to expand on this you could add "(unlike ESMF, you don't need to have the same mask for all data given to an initialised regridder)."

Reply:
20113ca5305a99af4e0e3e3ce76d83729dfe9705